src/app/shared/modules/create-product-modal/create-product-modal.component.ts

```diff
@@ -130,8 +158,9 @@ export class CreateProductModalComponent implements OnDestroy {
     return (
       this.form.invalid ||
       !this.formValues ||
-      Object.values(this.formValues).filter(Boolean).length !==
-        Object.values(this.form.value).length
+      Object.values(this.form.controls).some((field) =>
```

Comment:
The button should be disabled only when *this.form.valid* is false - if this is not working - then probably form validators are not set correctly according to task requirements. Please check and update.